astropy/units/tests/test_quantity_non_ufuncs.py

```diff
@@ -2578,3 +2585,170 @@ def test_all_included(self):
+
+            pos_target += 1
+
+    def test_known_arguments(self, functions):
+        # validate that all exposed arguments map to something in the target
+        target, helper = functions
+
+        try:
+            sig_target = inspect.signature(target)
+        except ValueError:
+            pytest.skip("Non Python function cannot be inspected at runtime")
+
+        params_target = sig_target.parameters
+        sig_helper = inspect.signature(helper)
+        params_helper = sig_helper.parameters
+
+        # a : positional-only
+        positionals_only_target = self._get_param_group(
+            params_helper, [POSITIONAL_ONLY]
+        )
+        positionals_only_helper = self._get_param_group(
+            params_helper, [POSITIONAL_ONLY]
+        )
+
+        if (nhelper := len(positionals_only_helper)) > (
+            ntarget := len(positionals_only_target)
+        ):
+            unknown: list[str] = positionals_only_helper[ntarget:]
+            raise AssertionError(
+                "Found unknown positional-only parameter(s) in helper's signature: "
+                f"{unknown}, at position(s) {list(range(ntarget, nhelper))}"
+            )
+
+        # b: positional-or-keyword
+        # TODO: reduce code duplication with pytest.mark.parametrize ?
+        positional_or_keyword_target = self._get_param_group(
+            params_target, [POSITIONAL_OR_KEYWORD]
+        )
+        positional_or_keyword_helper = self._get_param_group(
+            params_helper, [POSITIONAL_OR_KEYWORD]
+        )
+
+        if (nhelper := len(positional_or_keyword_helper)) > (
+            ntarget := len(positional_or_keyword_target)
+        ):
+            unknown: list[str] = positional_or_keyword_helper[ntarget:]
+            raise AssertionError(
+                "Found unknown positional-or-keyword parameter(s) in helper's signature: "
+                f"{unknown}, at position(s) {list(range(ntarget, nhelper))}"
+            )
+
+        # c: keyword-allowed
+        keyword_allowed_target = set(
+            self._get_param_group(params_target, [KEYWORD_ONLY, POSITIONAL_OR_KEYWORD])
+        )
+        keyword_allowed_helper = set(
+            self._get_param_group(params_helper, [KEYWORD_ONLY, POSITIONAL_OR_KEYWORD])
+        )
+
+        # additional private keyword-only argument are allowed because
+        # they are only intended for testing purposes.
+        keyword_allowed_helper = {
+            name for name in keyword_allowed_helper if not name.startswith("_")
+        }
+
+        assert not (diff := keyword_allowed_helper - keyword_allowed_target), (
+            "Found some keyword-allowed parameters in helper "
+            f"that are unknown to target: {diff}"
+        )
+
+        # finally, check that default values are correctly replicated
+        for name, ph in params_helper.items():
+            if name not in params_target:
+                # in a few select cases, this is okay,
```

Comment:
Is this still needed?

Reply:
yes. This is how we get pass `*args` or `**kwargs`, or additional private keyword-only arguments being present in some helper functions' signature when it's not also in the target signature. As I meant to express in the comment, this part intentionally doesn't replicate the exact logic to keep things clean with respect to the separation of concern principle. Let me know if I should rephrase to make it clearer !

Reply:
I think it would be clearer to have the explicit examples of cases where it is OK. Eg.,
```
# In a few cases, the helper defines names that are not in
# the target (e.g., a private name like _q_unit in quantile,
# or a *args, **kwargs that captures further arguments
# that do not matter. We let such cases slip by.
```